### ISSUE TITLE
Ability to exclude/include the platform .NET Core

### ISSUE THREAD

**Issue by jnm2** [nunit/nunit#2432]
The .NET Standard tests may be executed against a variety of runtimes, so ideally I would write this:

```cs
[Test]
[Platform(Exclude = "netcore", Reason = "BeginInvoke causes PlatformNotSupportedException")]
public static void WarningInBeginInvoke()
{
    new Action(() => Assert.Warn("(Warning message)"))
        .BeginInvoke(null, null)
        .AsyncWaitHandle.WaitOne();
}
```

Using `#if !(NETSTANDARD1_3 || NETSTANDARD1_6)` is not ideal because the .NET Standard tests can (and in theory should) be run against .NET Framework and Mono as well.

Unless we've got a new include/exclude attribute lined up to implement, can we just add the platform to the existing platform attribute?

**Comment by CharliePoole** [comment]
I strongly favor completely replacing the Platform and Culture attributes.   But there is no harm in adding this now if you can detect .net core.

**Comment by CharliePoole** [comment]
If anyone wants to work on this, there's a first cut at an implementation at https://github.com/CharliePoole/nunit in the platform-attr branch.

Currently all the (simulated) detection tests pass but there are some problems with current platform detection, which is necessary for actual use. I think the 1.6 implementation works pretty well but 1.3 is problematic. I won't be doing anything on this for the next month, so feel free to copy and use if you want to work this issue.

**Comment by ChrisMaddock** [comment]
Some updates to this issue for anyone willing to offer a PR:
1) Note I think this issue is really two part. First the existing platform attribute needs to be enabled on the .NET Standard platforms, secondly, the new "netcore" platform needs to be added. I think this makes sense staying as one feature however - as the first part has little practical use without the second.
2) Charlie refers to .NET Standard 1.3 and 1.6 above - we no longer target either of those platforms. Instead we now target .NET Standard 1.4 and 2.0. Ideally of course, the attribute would be available on both platforms, but I don't know what APIs are available in 1.4 to do this - so it make work out that only making the platform available in 2.0 is the best solution.

Note this attribute, and all related behaviour is currently wrapped up in a PLATFORM_DETECTION pre-compilation symbol, which is currently excluded from both .NET Standard builds [here](https://github.com/nunit/nunit/blob/cefc0c3a86c065daa4130aa255b1384db51b2b08/src/NUnitFramework/Directory.Build.props).
